Modules/posixmodule.c

```diff
@@ -5088,212 +5088,242 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
+#define PY_IFREG 1
+#define PY_IFDIR 2
+#define PY_IFLNK 3
+#define PY_IFMNT 4
+
+static BOOL
+_testFileTypeByHandle(HANDLE hfile, int refFileType, BOOL diskOnly)
+{
+    assert(refFileType == PY_IFREG || refFileType == PY_IFDIR ||
+           refFileType == PY_IFLNK || refFileType == PY_IFMNT);
+
+    DWORD fileDevType = GetFileType(hfile);
+    if ((fileDevType == FILE_TYPE_CHAR) ||
+        (fileDevType != FILE_TYPE_DISK && diskOnly) ||
+        (fileDevType == FILE_TYPE_UNKNOWN && GetLastError() != 0))
+    {
+        return FALSE;
+    }
+
+    FILE_INFO_BY_HANDLE_CLASS FileInformationClass;
+    LPVOID info;
```

Comment:
You need separate `FILE_ATTRIBUTE_TAG_INFO atInfo` and `FILE_BASIC_INFO baInfo` records. The set `pInfo` to one or the other based on the info class.

Reply:
I suggest using the camel-case name `fileInfoClass` instead of `FileInformationClass`.

Reply:
OK, but that's the name Microsoft uses: https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getfileinformationbyhandleex#syntax